src/awkward/_v2/forms/__init__.py

```diff
@@ -0,0 +1,121 @@
+        )
+    if "ListArray" == input["class"]:
+        return awkward._v2.forms.listform.ListForm(
+            starts=input["starts"],
+            stops=input["stops"],
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "ListOffsetArray" == input["class"]:
+        return awkward._v2.forms.listoffsetform.ListOffsetForm(
+            offsets=input["offsets"],
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "IndexedArray" == input["class"]:
+        return awkward._v2.forms.indexedform.IndexedForm(
+            index=input["index"],
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "IndexedOptionArray" == input["class"]:
+        return awkward._v2.forms.indexedoptionform.IndexedOptionForm(
+            index=input["index"],
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "IndexedOptionArray" == input["class"]:
+        return awkward._v2.forms.indexedoptionform.IndexedOptionForm(
+            index=input["index"],
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "ByteMaskedArray" == input["class"]:
+        return awkward._v2.forms.bytemaskedform.ByteMaskedForm(
+            mask=input["mask"],
+            content=from_iter(input["content"]),
+            valid_when=input["valid_when"],
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "BitMaskedArray" == input["class"]:
+        return awkward._v2.forms.bitmaskedform.BitMaskedForm(
+            mask=input["mask"],
+            content=from_iter(input["content"]),
+            valid_when=input["valid_when"],
+            lsb_order=input["lsb_order"],
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "UnmaskedArray" == input["class"]:
+        return awkward._v2.forms.unmaskedform.UnmaskedForm(
+            content=from_iter(input["content"]),
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
+    if "VirtualArray" == input["class"]:
+        return awkward._v2.forms.virtualform.VirtualForm(
+            form=input["form"],
+            has_length=input["has_length"],
+            has_identities=has_identities,
+            parameters=parameters,
+            form_key=form_key,
+        )
```

Comment:
If the above `if` statements are rearranged into an `if`-`elif`-`elif`-`else` chain, then the `else` case can raise a ValueError saying that `input["class"]` wasn't recognized.